utils/parametric/protos/apm_test_client.proto

```diff
@@ -37,12 +38,22 @@ message StartSpanArgs {
   optional string type = 5;
   optional string origin = 6;
   optional DistributedHTTPHeaders http_headers = 7;
+  repeated SpanLinks span_links = 8;
 }
 
 message DistributedHTTPHeaders {
   repeated HeaderTuple http_headers = 9;
 }
 
+message SpanLinks {
+  repeated SpanLink links = 1;
+}
+
+message SpanLink {
+  optional uint64 parent_id = 1;
```

Comment:
```suggestion
  uint64 trace_id = 1;
  uint64 parent_id = 2;
```

Reply:
This endpoint doesn't test setting an arbitrary trace_id/span_id, but:

> create a direct link between two spans given two valid span (or SpanContext) objects.

So, it's really referencing a known span from another span by its span_id.

Once you have that span object or span context object, the parametric test wants you to call the `span->add_link_from_span(otherSpan)` method. Or the `span->add_link(otherSpan->getLink())` methods, depending on the specific APIs you have.

Reply:
Okay I see what you're saying, but could we maybe add trace_id as optional? I'd like to test that we can accept an arbitrary trace ID and span ID, since the public API will allow for that, IIRC.

Reply:
There's no requirement to support that directly, it's part of the distributed tracing support - transforming the distributed headers into a span link, if it's from external, or just a getLink() from an internal one.

The primary reason being that span context should not really be propagated without format (datadog, tracecontext, whatever), otherwise you're going to be concerned with ... input as 128 bit hex... 64 bit hex, 64 bit dec? Keeps things simpler.